api/src/domain/user/taxEncryptionDecryption.test.ts

```diff
@@ -0,0 +1,116 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import * as AWSCrypto from "@aws-crypto/client-node";
+import { TextEncoder } from "node:util";
+import { EncryptionDecryptionClient } from "../types";
+import { TaxEncryptionDecryptionFactory } from "../user/taxEncryptionDecryptionFactory";
+
+type context = {
+  stage: string;
+  purpose: string;
+  origin: string;
+};
+
+type MockAWSCryptoType = {
+  generatorKeyIdCalledWith: string;
+  encryptValueCalledWith: string;
+  decryptValueCalledWith: string;
+  optionsCalledWith: string;
+  contextCalledWith: any;
+  buildClient: (options: string) => {
+    encrypt: (cmm: AWSCrypto.KeyringNode, plaintext: string, op?: context) => Promise<any>;
+    decrypt: (keyring: AWSCrypto.KeyringNode, value: string) => Promise<any>;
+  };
+  KmsKeyringNode: () => void;
+  CommitmentPolicy: any;
+};
+
+type MockAWSBase64 = {
+  fromBase64: (value: string) => string;
+  toBase64: (value: string) => string;
+};
+
+const mockAWSCrypto = AWSCrypto as unknown as jest.Mocked<MockAWSCryptoType>;
+
+describe("TaxEncryptionDecryption", () => {
+  let client: EncryptionDecryptionClient;
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+    client = TaxEncryptionDecryptionFactory("123", [], {
+      stage: "string",
+      purpose: "string",
+      origin: "string",
+    });
+  });
+
+  it("calls encrypt with the correct value", async () => {
+    const result = await client.encryptValue("2323232");
+    expect(result).toEqual("2323232");
+    expect(mockAWSCrypto.encryptValueCalledWith).toEqual("2323232");
+    expect(mockAWSCrypto.contextCalledWith).toEqual({
+      encryptionContext: {
+        stage: "string",
+        purpose: "string",
+        origin: "string",
+      },
+    });
+    expect(mockAWSCrypto.optionsCalledWith).toEqual("REQUIRE_ENCRYPT_REQUIRE_DECRYPT");
+  });
+
+  it("calls decrypt with the correct value", async () => {
+    await client.decryptValue("2323232");
+    expect(mockAWSCrypto.decryptValueCalledWith).toEqual("2323232");
+  });
+});
```

Comment:
there is a missing expectation in both tests that doesn't test what either of these functions _returns_